src/Expander.hs

```diff
@@ -914,13 +905,20 @@ runTask (tid, localData, task) = withLocal localData $ do
           case ty of
             Ty (TyF (TMetaVar ptr) _) | ptr == ptr' -> stillStuck tid task
             _ -> forkAwaitingTypeCase loc dest (tMetaVar ptr') env cases kont
-        other -> do
-          selectedBranch <- expandEval $ withEnv env $ doTypeCase loc (Ty other) cases
-          case selectedBranch of
+        other -> case doTypeCase env loc (Ty other) cases of
+          Right v -> case v of
             ValueMacroAction nextStep -> do
               forkInterpretMacroAction dest nextStep kont
             otherVal -> do
-              expandEval $ evalErrorType "macro action" otherVal
+              p <- currentPhase
+              throwError $ MacroEvaluationError p $ evalErrorType "macro action" otherVal
+          Left err -> do
+            p <- currentPhase
+            throwError
+              $ MacroEvaluationError p
+              $ evalErrorType "macro action"
```

Comment:
:x: this isn't a type error, this is converting the evaluation error into an `Expand` error.